test/unittest/rtps/security/SecurityTests.cpp

```diff
@@ -39,8 +39,10 @@ void SecurityTest::initialization_ok()
             WillOnce(DoAll(SetArgPointee<0>(stateless_reader_), Return(true))).
             WillOnce(DoAll(SetArgPointee<0>(volatile_reader_), Return(true)));
 
-    ASSERT_TRUE(manager_.init(security_attributes_, participant_properties_, security_activated_));
-    ASSERT_TRUE(!security_activated_ || manager_.create_entities());
+    security_activated_ = manager_.init(security_attributes_, participant_properties_);
+    ASSERT_TRUE(security_activated_);
+    ASSERT_TRUE(manager_.is_security_initialized());
+    ASSERT_TRUE(!manager_.is_security_initialized() || manager_.create_entities());
```

Comment:
```suggestion
    ASSERT_TRUE(manager_.create_entities());
```
`!manager_.is_security_initialized()` can't evaluate `true` or else the previous ASSERT would fail and abort the test.

Reply:
right